### PR TITLE
[Push] Add push messages history getter, add account to subscription

### PR DESCRIPTION
**Pull request by llbartekll**
according to spec change
https://github.com/WalletConnect/walletconnect-docs/pull/416/files

### CODE REVIEW

**Review by alexander-lsvk — APPROVED** [review]

File: Sources/WalletConnectPush/Client/Wallet/WalletPushClient.swift [hunk 1]
```diff
@@ -45,6 +46,7 @@ public class WalletPushClient {
          proposeResponder: PushRequestResponder,
          pushMessageSubscriber: PushMessageSubscriber,
          subscriptionsProvider: SubscriptionsProvider,
+         pushMessagesProvider: PushMessagesProvider,
          deletePushSubscriptionService: DeletePushSubscriptionService,
          deletePushSubscriptionSubscriber: DeletePushSubscriptionSubscriber,
          resubscribeService: PushResubscribeService) {
```
Comment (flypaper0) [inline Sources/WalletConnectPush/Client/Wallet/WalletPushClient.swift:52]: resubscribeService unused in PushClient

File: Sources/WalletConnectPush/Client/Wallet/PushMessagesProvider.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+import Foundation
+import WalletConnectUtils
+
+class PushMessagesProvider {
```
Comment (flypaper0) [inline Sources/WalletConnectPush/Client/Wallet/PushMessagesProvider.swift:5]: 👍

File: Sources/WalletConnectUtils/RPCHistory/RPCHistory.swift [hunk 3]
```diff
@@ -65,6 +65,16 @@ public final class RPCHistory {
         }
     }
 
+    public func getAll<Object: Codable>(of type: Object.Type, topic: String) -> [Object] {
+        return storage.getAll()
+            .filter{$0.topic == topic}
+            .compactMap { record in
```
Comment (flypaper0) [inline Sources/WalletConnectUtils/RPCHistory/RPCHistory.swift:71]: `.compactMap { try? $0.request.params?.get(Object.self) }`

**Review by flypaper0 — APPROVED** [review]

File: Sources/WalletConnectPush/Client/Wallet/WalletPushClient.swift (hunk 1 above)
Comment (llbartekll) [reply Sources/WalletConnectPush/Client/Wallet/WalletPushClient.swift:52]: it is in use, it's calling setUpResubscription func after initialisation

**Pull request merged**